scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1604,8 +1603,13 @@
     (prepare-match-trigger-chan match-trigger-chan pools')
     (async/go-loop []
       (when-let [x (async/<! match-trigger-chan)]
-        (let [iterator (swap! pool-name-iterator-atom #(if (.hasNext %) % (.iterator pool-names-array-list)))]
-          (async/offer! (get pool->match-trigger-chan (.next iterator)) x))
+        (try
+          (let [pool-name (.removeFirst pool-names-linked-list)
+                _ (.addLast pool-names-linked-list pool-name)]
```

Comment:
Why not move this line to after the bindings?

Reply:
updated